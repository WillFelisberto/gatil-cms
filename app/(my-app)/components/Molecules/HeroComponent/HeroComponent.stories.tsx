import type { Meta, StoryObj } from '@storybook/react';

import type { BlockProps } from '@/app/(my-app)/types/block-props';

import { HeroComponent } from './HeroComponent';

const meta: Meta<typeof HeroComponent> = {
  title: 'Molecules/HeroComponent',
  component: HeroComponent,
  tags: ['autodocs'],
  argTypes: {
    title: { control: 'text' },
    subtitle: { control: 'text' },
    ctaText: { control: 'text' },
    ctaLink: { control: 'text' }
  }
};

export default meta;
type Story = StoryObj<typeof HeroComponent>;

export const Default: Story = {
  args: {
    title: 'Ajude um Gato Hoje',
    subtitle: 'Cada produto comprado ajuda um resgatado üêæ',
    ctaText: 'Conhe√ßa os produtos',
    ctaLink: '/produtos',
    backgroundImage: {
      url: '/cat-hero.jpg'
    },
    blockType: 'hero'
  } as BlockProps<'hero'>
};
