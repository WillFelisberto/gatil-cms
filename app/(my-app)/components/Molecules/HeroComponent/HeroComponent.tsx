import Link from 'next/link';

import { BlockProps } from '@/app/(my-app)/types/block-props';
import { Media } from '@/payload-types';

export const HeroComponent = (props: BlockProps<'hero'>) => {
  const { title, subtitle, backgroundImage, ctaText, ctaLink } = props;

  const bgImage = (backgroundImage as Media)?.url;

  return (
    <section
      data-testid="hero-block"
      className="flex items-center justify-center text-white w-full min-h-screen h-screen overflow-hidden"
    >
      {/* Background Image */}
      {bgImage && (
        <div className="absolute inset-0 -z-10">
          <img
            src={bgImage}
            alt="Imagem de fundo"
            className="w-full h-full object-cover object-center"
          />
          <div className="absolute inset-0 bg-black/50" />
        </div>
      )}

      {/* Content */}
      <div className="text-center px-4 max-w-2xl mx-auto">
        <h1 className="text-4xl md:text-6xl font-extrabold mb-4" data-testid="hero-title">
          {title}
        </h1>
        {subtitle && (
          <p className="text-lg md:text-xl text-white/90 mb-6" data-testid="hero-subtitle">
            {subtitle}
          </p>
        )}
        {ctaText && ctaLink && (
          <Link
            href={ctaLink}
            data-testid="hero-cta"
            className="inline-block bg-pink-600 hover:bg-pink-700 text-white font-semibold text-base px-6 py-3 rounded-xl transition-all"
          >
            {ctaText}
          </Link>
        )}
      </div>
    </section>
  );
};
