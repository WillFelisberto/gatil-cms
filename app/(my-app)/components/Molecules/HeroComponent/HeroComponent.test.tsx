import type { BlockProps } from '@/app/(my-app)/types/block-props';
import { render, screen } from '@/tests/test-utils';

import { HeroComponent } from './HeroComponent';

const defaultProps: BlockProps<'hero'> = {
  title: 'Adote um Gato',
  subtitle: 'Você pode mudar uma vida hoje.',
  backgroundImage: {
    id: '1',
    createdAt: '2025-04-02T20:48:51.116Z',
    updatedAt: '2025-04-02T20:48:51.116Z',
    alt: 'Will',
    filename: 'gato3.png',
    mimeType: 'image/png',
    filesize: 1008532,
    width: 800,
    height: 800,
    focalX: 50,
    focalY: 50,
    sizes: {
      thumbnail: {
        width: 400,
        height: 300,
        mimeType: 'image/png',
        filesize: 324915,
        filename: 'gato3-400x300.png',
        url: '/api/media/file/gato3-400x300.png'
      },

      card: {
        width: 768,
        height: 1024,
        mimeType: 'image/png',
        filesize: 1794522,
        filename: 'gato3-768x1024.png',
        url: '/api/media/file/gato3-768x1024.png'
      },

      tablet: {
        width: null,
        height: null,
        mimeType: null,
        filesize: null,
        filename: null,
        url: null
      }
    }
  },
  ctaText: 'Ver Gatos',
  ctaLink: '/adote',
  blockType: 'hero'
};

describe('<HeroComponent />', () => {
  it('Should renders the title and subtitle', () => {
    const { container } = render(<HeroComponent {...defaultProps} />);
    expect(screen.getByTestId('hero-block')).toBeInTheDocument();
    expect(screen.getByTestId('hero-title')).toHaveTextContent('Adote um Gato');
    expect(screen.getByTestId('hero-subtitle')).toHaveTextContent('Você pode mudar uma vida hoje.');

    expect(container).toMatchSnapshot();
  });

  it('Should renders CTA if link and text are provided', () => {
    render(<HeroComponent {...defaultProps} />);
    const button = screen.getByTestId('hero-cta');
    expect(button).toHaveTextContent('Ver Gatos');
    expect(button).toHaveAttribute('href', '/adote');
  });

  it('Should does not render CTA if missing text or link', () => {
    render(<HeroComponent {...defaultProps} ctaText={undefined} />);
    expect(screen.queryByTestId('hero-cta')).not.toBeInTheDocument();
  });
});
