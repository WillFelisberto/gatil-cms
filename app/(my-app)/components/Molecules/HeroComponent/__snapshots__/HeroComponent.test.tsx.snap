// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroComponent /> Should renders the title and subtitle 1`] = `
<div>
  <section
    class="relative flex items-center justify-center text-white h-[80vh] min-h-[500px] w-full overflow-hidden"
    data-testid="hero-block"
  >
    <div
      class="text-center px-4 max-w-2xl mx-auto"
    >
      <h1
        class="text-4xl md:text-6xl font-extrabold mb-4"
        data-testid="hero-title"
      >
        Adote um Gato
      </h1>
      <p
        class="text-lg md:text-xl text-white/90 mb-6"
        data-testid="hero-subtitle"
      >
        VocÃª pode mudar uma vida hoje.
      </p>
      <a
        class="inline-block bg-pink-600 hover:bg-pink-700 text-white font-semibold text-base px-6 py-3 rounded-xl transition-all"
        data-testid="hero-cta"
        href="/adote"
      >
        Ver Gatos
      </a>
    </div>
  </section>
</div>
`;
